tests/acceptance/bootstrap/GraphContext.php

```diff
@@ -2891,6 +2893,47 @@ public function userListsTheActivitiesOfSpaceUsingTheGraphApi(string $user, stri
 		$this->featureContext->setResponse($response);
 	}
 
+	/**
+	 * @When the public tries to check the activities of space :spaceName owned by user :user with password :password
+	 *
+	 * @param string $spaceName
+	 * @param string $user
+	 * @param string $password
+	 *
+	 * @return void
+	 */
+	public function thePublicTriesToCheckTheActivitiesOfSpaceOwnedByUserWithPassword(string $spaceName, string $user, string $password): void {
+		$response = GraphHelper::getActivities(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			"public",
+			$this->spacesContext->getSpaceIdByName($user, $spaceName),
+			$this->featureContext->getActualPassword($password)
+		);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @When /^the public tries to check the activities of (?:folder|file) "([^"]*)" from space "([^"]*)" owned by user "([^"]*)" with password "([^"]*)"/
+	 *
+	 * @param string $resource
+	 * @param string $space
+	 * @param string $owner
+	 * @param string $password
+	 *
+	 * @return void
+	 */
+	public function thePublicTriesToCheckTheActivitiesOfResourceFromSpaceOwnedByUserWithPassword(string $resource, string $space, string $owner, string $password) {
+		$response = GraphHelper::getActivities(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			"public",
+			$this->featureContext->getPasswordForUser($owner),
+			$this->spacesContext->getResourceId($owner, $space, $resource)
+		);
+		$this->featureContext->setResponse($response);
+	}
```

Comment:
@saw-jan does it possible to use one function for both step or it will go against our current practice

Reply:
good to have different step defs in this case